File: app/algorithms/caesar/caesar-decrypt.js
let pyodide;

async function init() {
    pyodide = await loadPyodide();
    const response = await fetch("caesar-decrypt.py");
    const code = await response.text();
    await pyodide.runPythonAsync(code);
}

document.addEventListener("DOMContentLoaded", () => {
    init();

    document.getElementById("decrypt-btn").addEventListener("click", async () => {
        const text = document.getElementById("ciphertext").value;
        const shift = parseInt(document.getElementById("shift").value);

        pyodide.globals.set("ciphertext", text);
        pyodide.globals.set("shift", shift);

        const result = await pyodide.runPythonAsync(`decrypt(ciphertext, shift)`);
        document.getElementById("output").textContent = result;
    });
});

File: app/algorithms/caesar/caesar-decrypt.py
def encrypt(plaintext, shift):
    ciphertext = ""

    for i in range(len(plaintext)):
        char = plaintext[i]

        if (char.isupper()):
            ciphertext += chr((ord(char) + shift-65) % 26 + 65)
        
        else:
            ciphertext += chr((ord(char) + shift - 97) % 26 + 97)
    
    return ciphertext

def decrypt(ciphertext, shift):
    return encrypt(ciphertext, -shift)